packages/lib/services/e2ee/crypto.ts

```diff
@@ -0,0 +1,132 @@
+import { _ } from '../../locale';
+import { Crypto, CryptoBuffer, Digest, CipherAlgorithm, EncryptionResult } from './types';
+import { promisify } from 'util';
+import {
+	getCiphers as nodeGetCiphers, getHashes as nodeGetHashes,
+	randomBytes as nodeRandomBytes,
+	pbkdf2 as nodePbkdf2,
+	createCipheriv, createDecipheriv,
+	CipherGCMOptions, CipherGCM, DecipherGCM,
+} from 'crypto';
+
+const crypto: Crypto = {
+
+	getCiphers: () => {
+		return nodeGetCiphers();
+	},
+
+	getHashes: () => {
+		return nodeGetHashes();
+	},
+
+	randomBytes: async (size: number) => {
+		const randomBytesAsync = promisify(nodeRandomBytes);
+		return randomBytesAsync(size);
+	},
+
+	pbkdf2Raw: async (password: string, salt: CryptoBuffer, iterations: number, keylen: number, digest: Digest) => {
+		const pbkdf2Async = promisify(nodePbkdf2);
+		return pbkdf2Async(password, salt, iterations, keylen, digest);
+	},
+
+
+	encryptRaw: (data: CryptoBuffer, algorithm: CipherAlgorithm, key: CryptoBuffer, iv: CryptoBuffer | null, authTagLength: number, associatedData: CryptoBuffer | null) => {
```

Comment:
This and other methods appear to be private functions - used only from within this file. Please move all these functions out of the interface, and makes them simple (non-exported) functions. Also update the interface to remove all private methods.

Reply:
Modified in commit d780b09002adc955be9a7f3557cc54c8adec7276